pyproject.toml

```diff
@@ -0,0 +1,168 @@
+# To run the tests for all supported and installed python version:
+# on Fedora 37, run: sudo dnf install tox, poetry && tox -r -epy27 -epy36 # or no args
+# Quick overview: https://pypi.org/project/poetry/
+# On Fedora, tox.rpm installs all Python versions, otherwise use pyenv to install them:
+# https://brandonrozek.com/blog/pyenvtox/
+# https://realpython.com/intro-to-pyenv/
+
+[tool.poetry]
+# Documentation: https://python-poetry.org/docs/pyproject
+name = "python-libs"
+version = "0.1.0"
+description = "Common XCP-ng Python classes"
+license = "BSD-2-Clause"
+authors = [
+    "Simon Rowe",
+    "Andrew Cooper",
+    "Yann Dirson",
+]
+maintainers = [
+    "Ross Lagerwall",
+    "Pau Ruiz Safont",
+    "Bernhard Kaindl",
+]
+readme = "README"
+homepage = "https://github.com/xenserver/python-libs/"
+repository = "https://github.com/xenserver/python-libs/"
+keywords = ["xcp-ng", "xen-project", "libraries"]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python :: 2.7",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: System :: Hardware",
+    "Topic :: System :: Installation/Setup",
+    "Topic :: System :: Networking",
+    "Topic :: System :: Systems Administration",
+    "Topic :: Software Development :: Libraries :: Python Modules"
+]
+# https://python-poetry.org/docs/pyproject/#packages
+packages = [{ include = "xcp" }]
+
+[tool.poetry.urls]
+"Bug Tracker" = "https://github.com/xenserver/python-libs/issues"
+
+# Whenever you change dependencies by hand in your pyproject.toml,
+# you have to take care of these points
+# Run poetry lock --no-update afterwards. The reasons for this is, that poetry install
+# takes the poetry.lock as input if can find one and not the pyproject.toml.
+# If you change the python version and uses in-project virtualenv, remove the .venv
+# before running poetry install. poetry doesn't change the python version of a venv
+# once it is created, because it uses the python version itself to create the virtualenv.
+[tool.poetry.dependencies]
+# https://python-poetry.org/docs/dependency-specification/
+configparser = { version = "^4.0.2", python = "~2.7" }
```

Comment:
This is a library, I recommend against locking libraries to a single version. This should only done for binaries / applications